modules/@apostrophecms/doc-type/index.js

```diff
@@ -1757,6 +1758,32 @@ module.exports = {
               add.push('metaType');
             }
 
+            // Add relationships storage fields in projection
+            if (
+              relationships &&
+              Object.keys(projection).length &&
+              !Object.values(projection).some((val) => !val)
+            ) {
+              // Didn't find any other way, should work
+              const type = query.get('type');
+              const manager = self.apos.doc.getManager(type);
+              const directRelations = Array.isArray(relationships)
+                ? relationships.map(rel => rel.split('.')[0])
+                : [];
+
+              if (manager) {
+                for (const field of manager.schema) {
+                  if (field.type !== 'relationship') {
+                    continue;
+                  }
+
+                  if (relationships === true || directRelations.includes(field.name)) {
+                    add.push(field.idsStorage);
```

Comment:
What about `fieldsStorage`? Make sure to add it only if it is truthy.

Reply:
Thanks! Added!